pyspades/weapon.py

```diff
@@ -101,14 +138,47 @@ def get_ammo(self, no_max: bool = False) -> int:
             return ammo
         return max(0, ammo)
 
-    def is_empty(self, tolerance=CLIP_TOLERANCE) -> bool:
-        return self.get_ammo(True) < -tolerance or not self.shoot
+    def is_empty(self, tolerance: float = CLIP_TOLERANCE) -> bool:
+        """
+        Is this weapon empty?
+        """
+        # FIXME: This appears to be more of a "can this gun fire?" check.
+        # We may need to rename it. --GM
+        return self.get_ammo(no_max=True) < -tolerance or not self.shoot
 
-    def get_damage(self, value, position1, position2) -> int:
+    @abstractmethod
+    def get_damage(self, value: int, position1: Any, position2: Any) -> int:
+        """
+        Returns the amount of damage dealt by this weapon.
+
+        value: An index of the body part we're hitting.
+        position1: The starting position of the shot.
+        position2: The position of the target we're hitting.
+        """
+        raise NotImplementedError()
```

Comment:
Abstract methods on ABCs that are not overriden raise automatically on class creation.